tony-core/src/main/java/com/linkedin/tony/util/VersionInfo.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+
+package com.linkedin.tony.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+
+import static com.linkedin.tony.TonyConfigurationKeys.*;
+
+
+/**
+ * This class returns build information by reading the version-info.properties file.
+ */
+public class VersionInfo {
+  private static final Log LOG = LogFactory.getLog(VersionInfo.class);
+
+  private Properties versionInfo;
+
+  private VersionInfo() {
+    versionInfo = new Properties();
+    String versionInfoFile = "version-info.properties";
+    try {
+      InputStream is = Thread.currentThread().getContextClassLoader()
```

Comment:
why throw exception and then catch exception and print a log? Can we just if (is == null ) {log; return;}?

Reply:
I think it was written this way to avoid duplicate log lines. The current way it's written, there's just one log line on line 38.

Reply:
oh i see.. I feel it is more intuitive to just log instead of throwing an exception. Also we can use try-with-resources

```
try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream(versionInfoFile)){
if (is == null) {
   LOG
} else {
 versionInfo.load(is);
} catch () {
}
```

Also it looks phishing that getResourceAsStream would return null & throw IOException both. I'd suppose that function should either throw exception or return null?

Reply:
Updated to use try-with-resources and just log instead of throw exception if `is` is `null`.